flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/Execution.java

```diff
@@ -1248,21 +1251,9 @@ private void processFail(Throwable t, boolean isCallback, Map<String, Accumulato
 
 				releaseAssignedResource(t);
 				vertex.getExecutionGraph().deregisterExecution(this);
-				handlePartitionCleanup(releasePartitions, releasePartitions);
 
-				if (!isCallback && (current == RUNNING || current == DEPLOYING)) {
-					if (LOG.isDebugEnabled()) {
-						LOG.debug("Sending out cancel request, to remove task execution from TaskManager.");
-					}
-
-					try {
-						if (assignedResource != null) {
-							sendCancelRpcCall(NUM_CANCEL_CALL_TRIES);
-						}
-					} catch (Throwable tt) {
-						// no reason this should ever happen, but log it to be safe
-						LOG.error("Error triggering cancel call while marking task {} as failed.", getVertex().getTaskNameWithSubtaskIndex(), tt);
-					}
+				if (isLegacyScheduling()) {
+					maybeReleasePartitionsAndSendCancelRpcCall(current, isCallback, releasePartitions);
```

Comment:
Are we unifying these boolean flags in a separate issue?

Reply:
Yes. Let's do it in a separate issue.

Reply:
FLINK-14606 is opened for it.